### PR TITLE
use === and !== instead == and !=

### PR DESCRIPTION
**Pull request by popomore**
Split PR from #383

Based on #385 and #386, merge this PR after that.

### CODE REVIEW

**Comment by jlongster** [comment]
Same thing happened here, it's showing 3 commits for some reason.

**Comment by popomore** [comment]
It has been rebased.

File: src/compiler.js [hunk 1]
```diff
@@ -511,7 +511,7 @@ var Compiler = Object.extend({
             var name = target.value;
             var id = frame.lookup(name);
 
-            if (id == null) {
+            if (id === null || id === undefined) {
```
Comment (jlongster) [inline src/compiler.js]: I'm actually open to still using `==` for checking null/undefined. It gets annoying to always have to check for both, and once you learn that `==` matches both, I don't think its too confusing. That's the only case I'd want to use `==`. What do you think (others who see this, feel free to comment too)?

Comment (popomore) [inline src/compiler.js]: The practice leads us to always use `===` that `==` will confuse other what is exactly equal to.

But it works fine in this case.

**Comment by jlongster** [comment]
Thanks!

**Pull request merged**